### PR TITLE
Modify CreateAnew to check if new resource differs

### PR DESCRIPTION
**Pull request by tpantelis**
...from the existing one, if present. If not then don't delete and recreate. The check ignores the immutable metadata
fields and the `Status` field, if present.

### CODE REVIEW

**Review by dfarrell07 — APPROVED** [review]
LGTM, thanks as always @tpantelis!

File: pkg/util/create_or_update.go [hunk 1]
```diff
@@ -133,16 +134,27 @@ func CreateAnew(ctx context.Context, client resource.Interface, obj runtime.Obje
...
 	name := resource.ToMeta(obj).GetName()
 
-	var created runtime.Object
+	var retObj runtime.Object
 
 	err := wait.ExponentialBackoff(backOff, func() (bool, error) {
 		var err error
 
-		created, err = client.Create(ctx, obj, createOptions)
+		retObj, err = client.Create(ctx, obj, createOptions)
```
Comment (skitt) [inline pkg/util/create_or_update.go:142]: If this returns with no error, can’t we assume that the created object is correct, and return early?

Comment (tpantelis) [reply pkg/util/create_or_update.go:142]: That's the case on L144, ie `IsAlreadyExists` returns false if err is nil.

Comment (skitt) [reply pkg/util/create_or_update.go:142]: Ah yes, my brain saw the error-handling and didn’t spot that it applied even if there was no error!

**Review by skitt — APPROVED** [review]

**Pull request merged**